lib/lightning/attempts/attempt.ex

```diff
@@ -141,18 +145,26 @@ defmodule Lightning.Attempt do
     end)
   end
 
-  def complete(attempt, state) do
+  def complete(
+        attempt,
+        {state, error_type, _error_message} = _payload
+      ) do
     attempt
     |> cast(%{state: state}, [:state])
     |> validate_required([:state])
     |> validate_inclusion(:state, @final_states)
-    |> then(fn %{data: %{state: previous_state}} = changeset ->
-      if previous_state == :started do
+    |> then(fn %{data: %{state: previous_state, error_type: previous_error}} =
+                 changeset ->
+      if previous_state == :started and previous_error == nil do
         changeset
         |> change(finished_at: DateTime.utc_now())
+        |> change(error_type: error_type)
       else
         changeset
-        |> add_error(:state, "cannot complete attempt that is not started")
+        |> add_error(
+          :state,
+          "cannot complete attempt that is not started or has error"
+        )
```

Comment:
"or has a previous error" would make it more clear that it's regardless of payload `error_type` value